### DISCUSSION TITLE
External DHCP and multiple LTSP servers

### DISCUSSION THREAD

**Discussion by mecrider** [ltsp/ltsp#289] [Q&A]
We have been running LTSP5 on Ubuntu for years (currently 16.04), with 4 servers on our network for load balancing and redundancy. We use dnsmasq to run DHCP/DNS on a separate server, and use tags in the hostsfile and customized dhcp-boot/dhcp-option lines for each tag to designate which server a client boots from. Now that 20.04 is out, I have a test server up and running with the basic install steps, and can boot up a thick client and log in. However, I only got a successful boot by putting dnsmasq on the LTSP server in proxy mode, and all clients on the network boot to that server no matter what directions they have in the separate DHCP. Is it still possible to put commands in the external DHCP to boot to the new LTSP on a designated server, or is it possible to limit the machines on the same subnet that an LTSP server responds for?

**Comment by alkisg** [comment]
The best method is to configure the external DHCP server to only send IP information. To not send a boot filename at all.

Then the multiple LTSP servers will work as proxyDHCP servers.
And, to limit which server answers to which clients, you'd use something like the following:

```shell
# cd /etc/dnsmasq.d; ls
dhcp-hosts/        dhcp-hosts.conf    ltsp-dnsmasq.conf  README

# cat dhcp-hosts.conf 
dhcp-hostsfile=/etc/dnsmasq.d/dhcp-hosts
dhcp-ignore=tag:!lab1

# cat dhcp-hosts/lab1.conf 
08:00:27:29:13:01,set:lab1,pc01
08:00:27:34:13:02,set:lab1,pc02
```

And of course it's possible to not do proxyDHCP at all, and configure everything on your main server. For that, read ltsp-dnsmasq.conf, and try to imitate it.

**Comment by mecrider** [comment]
Thanks for pointing me in the right direction. Because our network has a fairly small number of clients for each LTSP server compared to the total number of devices served from our DHCP server, I changed dhcp-ignore to tag:!known, and in the last conf file I only listed the clients for that LTSP server, without setting any tags (e.g. 08:00:27:29:13:01,pc01). This is working perfectly.
